pkg/executor/simple_test.go

```diff
@@ -145,3 +147,74 @@ func TestRefreshStatsWhenDatabaseIsEmpty(t *testing.T) {
 	tk.MustExec("refresh stats test.*")
 	require.Len(t, vars.StmtCtx.GetWarnings(), 0)
 }
+
+func TestRefreshStatsPrivilegeChecks(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t_refresh_priv")
+	tk.MustExec("create table t_refresh_priv (a int)")
+
+	t.Run("table scope requires select", func(t *testing.T) {
+		tk.MustExec("drop user if exists 'refresh_reader'@'%'")
+		tk.MustExec("create user 'refresh_reader'@'%'")
+		t.Cleanup(func() {
+			tk.MustExec("drop user if exists 'refresh_reader'@'%'")
+		})
```

Comment:
Are these necessary? Or is not every test a new store including users etc.?

Reply:
In theory, they are not necessary because I used a different username. I deleted it.